### PR TITLE
Copter: 4.1.0-beta1 release notes

### PR DESCRIPTION
**Pull request by rmackay9**
Copter-4.1.0-beta release notes

I'm raising as a PR in order to get input from all the various devs because I'm sure I've forgotten some items.

Thanks for your help!

### CODE REVIEW

File: ArduCopter/ReleaseNotes.txt [hunk 1]
```diff
@@ -1,5 +1,83 @@
 ArduPilot Copter Release Notes:
 ------------------------------------------------------------------
+Copter 4.1.0-beta1 14-Apr-2021
+Changes from 4.0.7
+1) EKF changes:
+    a) EKF3 is default estimator (EKF2 is available as an option)
+    b) External AHRS/IMU support (e.g. VectorNav)
+    c) Gaussian Sum Filter (GSF) allows emergency yaw correction using GPS
+    d) GPS-for-yaw (dual F9 UBlox GPS can provide yaw)
```
Comment (WickedShell) [inline ArduCopter/ReleaseNotes.txt:9]:
```suggestion
    d) GPS-for-yaw (dual F9 UBlox, Septentrio, NMEA GPS can provide yaw)
```

**Review by WickedShell — COMMENTED** [review]
Other things that stand out from memory:
- `AUTO_OPTIONS` - Primarily because it allows you to arm in auto, as well as takeoff without raising the throttle
- Ability to mark a MAVLink channel as private so we don't forward data to/from it (I've been pinged by people asking about this in the stable release because they found it in param docs, so people do seem to care about it)
- Vibration is now logged for all IMU's allowing you to compare the reported vibration between primary and secondary IMU's
- GCS failsafe timeout is now a parameter, I don't think this was ever backported, and it seems hugely useful to me (but I implemented it so I'm probably biased)
- Support DO_LAND_START as a MAVLink command to jump to the landing sequence
- Supports UAVCAN GNSS Status, and broadcasts armed state over UAVCAN


This was sorta just a random troll through the PR's I had that seemed worthy of calling out as they may be of wider interest.

File: ArduCopter/ReleaseNotes.txt (hunk 1 above)
Comment (rmackay9) [reply ArduCopter/ReleaseNotes.txt:9]: great, thanks so much!

**Comment by andyp1per** [comment]
Harmonic notch improvements:

- Support for notch-per-motor with ESC telemetry and notch-per-peak with FFT (INS_HNTCH_OPTS = 2)
- Support for double-notch (INS_HNTCH_OPTS = 1)
- Notch slewing to avoid shot noise
- Increased notch update rate to avoid shot noise
- Support for in-flight FFT (FFT_* and INS_HNTCH_MODE = 4)
- Support for in-flight learning of throttle notch using in-flight FFT
- Support for bi-directional dshot
- Support for RPM status driven from harmonic notch input

Other improvements:
- Shared DMA now uses mutexes for improved locking performance
- Reduced invensense buffer sizes to avoid DMA contention
- Fully support half-duplex for protocols using 2 stop bits on UARTs
- Set FLTT to zero during autotune
- Support for air-mode via ACRO_OPTIONS
- Support for SRXL2 listen-only devices
- Support for scheduler introspection via @SYS/tasks.txt
- Support for DMA contention introspection via @SYS/dma.txt
- Support for UART send/receive introspection via @SYS/uart.txt

File: ArduCopter/ReleaseNotes.txt [hunk 2]
```diff
@@ -1,5 +1,83 @@
...
+    e) Lane switching logic improvements
+    f) Sensor affinity (improves sensor failure redundancy)
+    g) Source switching for GPS/Non-GPS transitions (see EK3_SRCx_ parameters)
+    h) Yaw estimation and reliability improvements
+2) Control and Navigation improvements:
+    a) SCurves for waypoint navigation
+    b) Auto mdoe arming and takeoff (see AUTO_OPTIONS)
```
Comment (lthall) [inline ArduCopter/ReleaseNotes.txt:16]: b) Auto mode arming and takeoff (see AUTO_OPTIONS)

Comment (lthall) [reply ArduCopter/ReleaseNotes.txt:16]: Add logging for 2D PID objects
Add Feed Forward to 2D PID object
Add slew limiting to AC_PID object
Update Position Controller

**Comment by tatsuy** [comment]
How about adding wind estimation and baro compensation #15774 to EKF changes?

File: ArduCopter/ReleaseNotes.txt (hunk 2 above)
Comment (IamPete1) [reply ArduCopter/ReleaseNotes.txt:16]: Added yaw imbalance check (https://github.com/ArduPilot/ardupilot/pull/15744)

File: ArduCopter/ReleaseNotes.txt [hunk 3]
```diff
@@ -1,5 +1,83 @@
...
+    m) Smart Audio
+    n) SMBus batteries up to 12 cells
+    o) USD1 CAN radar
+10) Scripting enhancements:
+    a) Button, Proximity, RPM sensor support
+    b) DO_ mission commands can be triggered from scripts
+    c) I2C sensor driver support (i.e. allows writing sensor drivers in Lua)
+    d) Logging (i.e. allows Lua scripts to write to onboard logs)
+    e) Mission item read support
+    f) Motor drivers support (used for 6DOF)
```
Comment (IamPete1) [inline ArduCopter/ReleaseNotes.txt:63]:
```suggestion
    f) Motor drivers support allowing custom frame types and 6DOF
```
https://github.com/ArduPilot/ardupilot/pull/15708

**Comment by IamPete1** [comment]
Not really sure where this one should go,

`Added circle options to allow circle mode to face direction of travel and/or init at circle center.`

(https://github.com/ArduPilot/ardupilot/pull/15286)

File: ArduCopter/ReleaseNotes.txt [hunk 4]
```diff
@@ -1,5 +1,83 @@
...
+3) 6DoF frame support
+4) Object avoidance:
+    a) BendyRuler hesitancy improvements
+    b) Intel Realsense 435/455 camera support (companion computer required)
+    c) Simple avoidance backs away from obstacles (see AVOID_BACKUP_SPD parameter)
+    d) Simple avoidance accel limited (see AVOID_ACCEL_MAX parameter)
+    e) Simultaneous Dijkstra and BendyRuler path planning
+    f) Obstacle database now 3D
+    g) Obstacle filtering improvements
+    h) Vertical BendyRuler
```
Comment (rishabsingh3003) [inline ArduCopter/ReleaseNotes.txt:26]: - Minimum altitude before Simple Avoidance is switched on (see AVOID_ALT_MIN parameter)

- Filter Obstacle near the ground using downward facing range finder (see PRX_IGN_GND parameter)

- Ignore obstacles near home in OA Database (see OA_DB_ALT_MIN)

- Proximity library supports 3D data (Through OBSTACLE_DISTANCE_3D mavlink message)

- Avoidance consumes 3D data (not sure how the above two should be worded, or if they should be even included. But it's a significant change!)

- Various Bug fixes (Like we added the face timeout, which was an issue in 4.0, and various issues were also fixed in Simple Avoidance)

**Comment by bnsgeyer** [comment]
@rmackay9 please add the following as Tradheli improvements
Improved conventional and compound helicopter SITL dynamic models
Added intermeshing rotor and coaxial rotor support for Dual heli frame

**Comment by andyp1per** [comment]
New boards:
- MambaF405v2
- i2c capable revo-mini
- Probably should mention all the the -bdshot firmware

**Comment by andyp1per** [comment]
Here's the complete list of @SYS:

   can0_stats.txt
   can1_stats.txt
   can_log.txt
   dma.txt
   memory.txt
   persistent.parm
   tasks.txt
   threads.txt
   uarts.txt

**Comment by rmackay9** [comment]
@andyp1per,

Thanks, I've added the Mamba boards but skipped adding more detail to the SYS stuff 'cuz I want each line to be short-ish and it's perhaps too much detail for most users.

By the way, the addition of the -bdshot firmware came into question with a partner who found that when they went to install 4.1 they were presented (by MP) with 3 options and they didn't know which to choose.

**Comment by andyp1per** [comment]
@rmackay9 maybe we should ask @meee1 to finesse the options presented?

**Comment by rmackay9** [comment]
I think these release notes are all ready to go now so I'll merge once it passes CI and push out the Copter-4.1.0-beta1 release tomorrow.  Thanks all!

**Pull request merged**